docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md

```diff
@@ -0,0 +1,287 @@
+1. When applicable, NuGet will iterate over the plugins in priority order and starts them one by one.
+1. NuGet will use the first plugin that can service the request.
+1. The plugins will be shut down when they are no longer needed.
+
+## General plugin requirements
+
+The current protocol version is *2.0.0*.
+Under this version, the requirements are as follows:
+
+- Have a valid, trusted Authenticode signature assemblies that will run on Windows and Mono. There is no special trust requirement for assemblies run on Linux and Mac yet. [Relevant issue](https://github.com/NuGet/Home/issues/6702)
+- Support stateless launching under the current security context of NuGet client tools. For example, NuGet client tools will not perform elevation or additional initialization outside of the plugin protocol described later.
+- Be non interactive, unless explicitly specified.
+- Adhere to the negotiated plugin protocol version.
+- Respond to all requests within a reasonable time period.
+- Honor cancellation requests for any in-progress operation.
+
+The technical specification is described in more detail in the following specs:
+
+- [NuGet Package Download Plugin](https://github.com/NuGet/Home/wiki/NuGet-Package-Download-Plugin)
+- [NuGet cross plat authentication plugin](https://github.com/NuGet/Home/wiki/NuGet-cross-plat-authentication-plugin)
+
+## Plugin installation and discovery
+
+The plugins will be discovered via a convention based directory structure.
+CI/CD scenarios and power users can use an environment variable to override the behavior.
+
+- `NUGET_PLUGIN_PATHS` - defines the plugins that will be used for that NuGet process, priority reserved. If this environment variable is set, it overrides the convention based discovery. 
+-  User-location, the NuGet Home location in `%UserProfile%/.nuget/plugins`. This location cannot be overriden. A different root directory will be used for .NET Core and .NET Framework plugins. 
+
+| Framework | Root discovery location  |
+| ------- | ------------------------ |
+| .NET Core |  `%UserProfile%/.nuget/plugins/netcore` |
+| .NET Framework | `%UserProfile%/.nuget/plugins/netfx` |
+
+Each plugin should be installed in its own folder.
+The plugin entry point will be the name of the installed folder, with the .dll extensions for .NET Core, and .exe extension for .NET Framework.
+
+```
+.nuget
+    plugins
+        netfx
+            myPlugin
+                myPlugin.exe
+                nuget.protocol.dll
+                ...
+        netcore
+            myPlugin
+                myPlugin.dll
+                nuget.protocol.dll
+                ...
+```
+
+The following table describes the correlation NuGet products and the type of plugin they support.
+
+| Client tool  | Framework |
+| ------------ | --------- |
+| Visual Studio | .NET Framework |
+| dotnet.exe | .NET Core |
+| NuGet.exe | .NET Framework |
+| MSBuild.exe | .NET Framework |
+| NuGet.exe on Mono | .NET Framework |

```

Comment:
lil confused with this table, so if I've a `netcore` plugin, will that not work in `Visual Studio`?

Reply:
Yup. 
Similar to NuGet.exe and MSBuild only supporting .NET Framework.